lapack/unit_test/Test_Lapack_svd.hpp

```diff
@@ -0,0 +1,321 @@
+    EXPECT_NEAR_KK_REL(sign_vec * Vt_h(0, 0), Vtref[0], 100 * eps);
+    EXPECT_NEAR_KK_REL(sign_vec * Vt_h(1, 0), Vtref[1], 100 * eps);
+    sign_vec = (Vt_h(0, 1).real() < KAT_S::zero().real()) ? mag_one : -mag_one;
+    EXPECT_NEAR_KK_REL(sign_vec * Vt_h(0, 1), Vtref[2], 100 * eps);
+    EXPECT_NEAR_KK_REL(sign_vec * Vt_h(1, 1), Vtref[3], 100 * eps);
+  } else {
+    mag_type sign_vec = (U_h(0, 0) > KAT_S::zero()) ? mag_one : -mag_one;
+    EXPECT_NEAR_KK_REL(sign_vec * U_h(0, 0), Uref[0], 100 * eps);
+    EXPECT_NEAR_KK_REL(sign_vec * U_h(1, 0), Uref[1], 100 * eps);
+    sign_vec = (U_h(0, 1) < KAT_S::zero()) ? mag_one : -mag_one;
+    EXPECT_NEAR_KK_REL(sign_vec * U_h(0, 1), Uref[2], 100 * eps);
+    EXPECT_NEAR_KK_REL(sign_vec * U_h(1, 1), Uref[3], 100 * eps);
+
+    sign_vec = (Vt_h(0, 0) > KAT_S::zero()) ? mag_one : -mag_one;
+    EXPECT_NEAR_KK_REL(sign_vec * Vt_h(0, 0), Vtref[0], 100 * eps);
+    EXPECT_NEAR_KK_REL(sign_vec * Vt_h(1, 0), Vtref[1], 100 * eps);
+    sign_vec = (Vt_h(0, 1) < KAT_S::zero()) ? mag_one : -mag_one;
+    EXPECT_NEAR_KK_REL(sign_vec * Vt_h(0, 1), Vtref[2], 100 * eps);
+    EXPECT_NEAR_KK_REL(sign_vec * Vt_h(1, 1), Vtref[3], 100 * eps);
+  }
+
+  return 0;
+}
+
+template <class AMatrix, class Device>
+int impl_test_svd(const int m, const int n) {
+  AMatrix A("A", m, n);
+
+  return 0;
+}
+
+}  // namespace Test
+
+template <class ScalarA, class Device>
+int test_svd() {
+  int ret;
+
+#if defined(KOKKOSKERNELS_INST_LAYOUTLEFT) || \
+    (!defined(KOKKOSKERNELS_ETI_ONLY) &&      \
+     !defined(KOKKOSKERNELS_IMPL_CHECK_ETI_CALLS))
+  using view_type_a_layout_left =
+      Kokkos::View<ScalarA**, Kokkos::LayoutLeft, Device>;
+
+  ret = Test::impl_analytic_svd<view_type_a_layout_left, Device>();
+
+  ret = Test::impl_test_svd<view_type_a_layout_left, Device>(0, 0);
+  EXPECT_EQ(ret, 0);
+
+  ret = Test::impl_test_svd<view_type_a_layout_left, Device>(1, 1);
+  EXPECT_EQ(ret, 0);
+
+  ret = Test::impl_test_svd<view_type_a_layout_left, Device>(15, 15);
+  EXPECT_EQ(ret, 0);
+
+  ret = Test::impl_test_svd<view_type_a_layout_left, Device>(100, 100);
+  EXPECT_EQ(ret, 0);
+#endif
+
+#if defined(KOKKOSKERNELS_INST_LAYOUTRIGHT) || \
+    (!defined(KOKKOSKERNELS_ETI_ONLY) &&       \
+     !defined(KOKKOSKERNELS_IMPL_CHECK_ETI_CALLS))
+  using view_type_a_layout_right =
+      Kokkos::View<ScalarA**, Kokkos::LayoutRight, Device>;
+
+  ret = Test::impl_analytic_svd<view_type_a_layout_right, Device>();
+
+  ret = Test::impl_test_svd<view_type_a_layout_right, Device>(0, 0);
+  EXPECT_EQ(ret, 0);
+
+  ret = Test::impl_test_svd<view_type_a_layout_right, Device>(1, 1);
+  EXPECT_EQ(ret, 0);
+
+  ret = Test::impl_test_svd<view_type_a_layout_right, Device>(15, 15);
+  EXPECT_EQ(ret, 0);
+
+  ret = Test::impl_test_svd<view_type_a_layout_right, Device>(100, 100);
+  EXPECT_EQ(ret, 0);
+#endif
+
+  return 1;
+}
+
+template <class Scalar, class Device>
+int test_wrapper() {
```

Comment:
Just one minor suggestion: can we change this to ```test_svd_wrapper```?

Reply:
Good point, that will reduce the likelihood of function redefinitions...